src/dex/swaap-v2/swaap-v2.ts

```diff
@@ -0,0 +1,713 @@
+          remaininig = remaininig.minus(levelDelta);
+          previousLevel = BigNumber(levels[i].level);
+          i += 1;
+        } else {
+          output = output.plus(remaininig.times(levels[i].price));
+          enoughLiquidity = true;
+          break;
+        }
+      }
+      if (enoughLiquidity) {
+        return BigInt(
+          output
+            .multipliedBy(
+              getBigNumberPow(
+                side === SwapSide.BUY ? srcToken.decimals : destToken.decimals,
+              ),
+            )
+            .toFixed(0),
+        );
+      }
+      return BigInt(0); // not enough liquidity
+    });
+  }
+
+  async getCachedLevels(): Promise<Record<string, SwaapV2PriceLevels> | null> {
+    const cachedLevels = await this.dexHelper.cache.get(
+      this.dexKey,
+      this.network,
+      getPriceLevelsCacheKey(this.dexKey),
+    );
+
+    if (cachedLevels) {
+      return JSON.parse(cachedLevels) as Record<string, SwaapV2PriceLevels>;
+    }
+
+    return null;
+  }
+
+  normalizeToken(token: Token): Token {
+    return {
+      address: normalizeTokenAddress(token.address),
+      decimals: token.decimals,
+    };
+  }
+
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<SwaapV2Data>> {
+    try {
+      const normalizedSrcToken = this.normalizeToken(srcToken);
+      const normalizedDestToken = this.normalizeToken(destToken);
+
+      const requestedPoolIdentifier: string = getPoolIdentifier(
+        this.dexKey,
+        normalizedSrcToken.address,
+        normalizedDestToken.address,
+      );
+
+      if (normalizedSrcToken.address === normalizedDestToken.address) {
+        return null;
+      }
+
+      const pools =
+        limitPools ??
+        (await this.getPoolIdentifiers(srcToken, destToken, side, blockNumber));
+
+      const levels = await this.getCachedLevels();
+      if (levels === null) {
+        return null;
+      }
+
+      const poolIdentifier = getPoolIdentifier(
+        this.dexKey,
+        normalizedSrcToken.address,
+        normalizedDestToken.address,
+      );
+
+      const levelEntries: SwaapV2PriceLevels[] = Object.keys(levels)
+        .map((pair: string) => {
+          if (pools.includes(poolIdentifier)) {
+            return levels[pair];
+          }
+          return undefined;
+        })
+        .filter((o): o is SwaapV2PriceLevels => o !== undefined);
+
+      const divider = getBigNumberPow(
+        (side === SwapSide.SELL ? normalizedSrcToken : normalizedDestToken)
+          .decimals,
+      );
+
+      const amountsFloat = amounts.map(a =>
+        new BigNumber(a.toString()).dividedBy(divider),
+      );
+
+      const prices = levelEntries.map((askAndBids: SwaapV2PriceLevels) => {
+
+        const unitPrice = this.computePricesFromLevelsBis(
+          [BN_1],
```

Comment:
weird that we're using 1 wei (no decimals computed), same for [hashflow](https://github.com/paraswap/paraswap-dex-lib/blob/a4669eb522a3b0805d676fe905aa2e0502e62a7d/src/dex/hashflow/hashflow.ts#L484) cc @Verisana

Reply:
@mwamedacen updated

Reply:
@aburkut , may you elaborate, please, what you updated? Here everything is good. Discussed in DM with @mwamedacen